uswds/uswds
# USWDS - Time picker: Remove initial input element

## Summary
Updated the time picker script to remove the initial input element. Removing the initial element resolves pa11y errors. 

### Breaking change
This is not a breaking change.

## Related issue
Closes https://github.com/uswds/uswds/issues/4954

## Preview link
Preview link: [Time picker component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-time-picker-pa11y/?path=/story/components-form-inputs-time-picker--time-picker)

## Problem statement
The hidden text input in the Time Picker does not pass accessibility scans with pa11y. 

```
Error: This textinput element does not have a name available to an accessibility API. Valid names are: label element, title undefined, aria-label undefined, aria-labelledby undefined.
[htmlcs: WCAG2AA.Principle4.Guideline4_1.4_1_2.H91.InputText.Name](https://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/#sc4_1_2)
<input type="text" class="usa-input" aria-describedby="appointment-time-label appointment-time-hint" style="display: none;">

Error: This form field should be labelled in some way. Use the label element (either with a "for" attribute or wrapped around the form field), or "title", "aria-label" or "aria-labelledby" attributes as appropriate.
[htmlcs: WCAG2AA.Principle1.Guideline1_3.1_3_1.F68](https://squizlabs.github.io/HTML_CodeSniffer/Standards/WCAG2/#sc1_3_1)
<input type="text" class="usa-input" aria-describedby="appointment-time-label appointment-time-hint" style="display: none;">
```

To see these errors, run the following in the command line:
```
pa11y 'https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html\?args\=\&id\=components-form-inputs-time-picker--time-picker\&viewMode\=story'
```

## Solution
Once initialized, the time picker component ends up with two input elements: the initial input and a newly generated input. 
This happens because time picker relies on `enhanceComboBox()`, which generates a new `input` element and adds needed attributes to that `input`. 

Currently, the initial input is hidden with "display:none" once initialized, which does not successfully hide the element from pa11y and results in errors. Because the initial `input` element is redundant, we should be able to safely remove it from the DOM.

![image](https://user-images.githubusercontent.com/93996430/197018971-36413f3c-6a73-4d99-a02d-921156e6b9a2.png)


## Testing and review
To test:
1. Run the preview page through pa11y by using the following in the command line, and confirm that no errors are reported:
`pa11y 'https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/al-time-picker-pa11y/iframe.html?args=&id=components-form-inputs-time-picker--time-picker&viewMode=story'` 
1. Confirm that nothing has changed in appearance, behavior, or assistive technology readout from the [current time picker component](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/?path=/story/components-form-inputs-time-picker--time-picker).

---

Before opening this PR, make sure you’ve done whichever of these applies to you:
